mdn/kuma
# Stripe webhook for sending invoices

Fixes #6562 

## What changed
- new task which send a payment received email, that looks like this:
<details>
  <summary>payment received email</summary>

```email
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: MDN invoice for 2020-03-10 09:19:40
From: notifications@developer.mozilla.org
To: gweber@mozilla.com
Date: Wed, 11 Mar 2020 12:11:36 -0000
Message-ID: <158392869624.10.12666483619108222079@e446061a0d47>

Hello from MDN!

https://pay.stripe.com/invoice/acct_1BGvg2IxHhlg31jE/invst_GswhYtgGTJ9xBsk44fDOyEkRA9jFX5g/pdf

Cancel any time. To cancel your subscription, please go to the manage monthly subscriptions page: http://localhost.org:8000/en-US/payments/recurring/management.
If you have questions, please read the FAQ (http://localhost.org:8000/en-US/payments/recurring/subscription) or you can also contact support: mdn-support@mozilla.com

Thank you for using MDN!
```
</details>

- new route for handling a stripe `invoice.payment_succeeded` webhook which runs the task mentioned above
- creates a webhook inside the configured stripe account on kuma startup, using the configured `DOMAIN` and the new webhook route


## How to test it
1. [Make sure you have stripe setup](https://kuma.readthedocs.io/en/latest/installation.html#enable-stripe-payments-optional)
2. Start Kuma
3. [Tunnel your local server](https://github.com/mdn/kuma/pull/6669/files#diff-548eb9ce4cac8e13a1238ad703272931)
4. Go to your profile edit page, cancel your subscription if you have one and create a new subscription
5. Navigate to `/app/tmp/emails` inside your docker to check if you received a payment mail


## Some possible points of contention
- I'm not sure if `UserConfig.ready` is the best place to create the missing stripe hook
- Should it be an HTML email instead?
